source/include/microapp/cs_MicroappController.h

```diff
@@ -9,6 +9,25 @@ extern "C" {
 
 static_assert(sizeof(bluenet2microapp_ipcdata_t) <= BLUENET_IPC_RAM_DATA_ITEM_SIZE);
 
+/**
+ * Runtime state of a microapp. For now binary, either running or not running.
+ */
+enum class MicroappRuntimeState {
+	CS_MICROAPP_NOT_RUNNING = 0,
+	CS_MICROAPP_RUNNING     = 1,
+};
+
+/**
+ * @struct MicroappRuntimeData
+ * @brief Runtime data across all microapps.
+ *
+ * @var MicroappRuntimeData::appState
+ *   State per microapp for at max MAX_MICROAPPS.
+ */
+struct MicroappRuntimeData {
+	MicroappRuntimeState appState[MAX_MICROAPPS];
+};
```

Comment:
If these structs change with a firmware upgrade, things might be interpreted wrong.
Maybe move this to the protocol dir or so?

Reply:
If there is a firmware update all the data in IPC ram should be cleared by the dfu process. Sounds like a much easier and reliable approach. It's not that we want to use part of the IPC data to communicate from the old firmware to the new firmware. :-)

I don't know what it has to do with protocol.

Reply:
Note that things change if there was some kind of general "RebootManager" that would hold state across reboots and if the MicroappController would be one of the entities that writes data to it. 

That wasn't the right approach and everything had to move into the MicroappController. So, it can do whatever it wants with that data. There's no protocol, because it knows how it reads what it writes.

Reply:
> If there is a firmware update all the data in IPC ram should be cleared by the dfu process. Sounds like a much easier and reliable approach. It's not that we want to use part of the IPC data to communicate from the old firmware to the new firmware. :-)
So this should be implemented in the bootloader then? Or did you have some other mechanism in mind?